crates/cheatcodes/src/inspector.rs

```diff
@@ -60,55 +61,91 @@ pub trait CheatcodesExecutor {
         };
 
-        let mut evm = new_evm_with_existing_context(inner, inspector);
+        let mut evm = new_evm_with_existing_context(inner, &mut inspector as _);
 
-        evm.context.evm.inner.journaled_state.depth += 1;
+        let res = f(&mut evm)?;
 
-        let first_frame_or_result =
-            evm.handler.execution().create(&mut evm.context, Box::new(inputs))?;
+        ccx.ecx.journaled_state = evm.context.evm.inner.journaled_state;
+        ccx.ecx.env = evm.context.evm.inner.env;
+        ccx.ecx.l1_block_info = evm.context.evm.inner.l1_block_info;
+        ccx.ecx.error = evm.context.evm.inner.error;
 
-        let mut result = match first_frame_or_result {
-            revm::FrameOrResult::Frame(first_frame) => evm.run_the_loop(first_frame)?,
-            revm::FrameOrResult::Result(result) => result,
-        };
+        Ok(res)
+    }
+
+    /// Obtains [revm::Evm] instance and executes the given CREATE frame.
+    fn exec_create<DB: DatabaseExt>(
+        &mut self,
+        inputs: CreateInputs,
+        ccx: &mut CheatsCtxt<DB>,
+    ) -> Result<CreateOutcome, EVMError<DB::Error>> {
+        self.with_evm(ccx, |evm| {
+            evm.context.evm.inner.journaled_state.depth += 1;
 
-        evm.handler.execution().last_frame_return(&mut evm.context, &mut result)?;
+            let first_frame_or_result =
+                evm.handler.execution().create(&mut evm.context, Box::new(inputs))?;
 
-        let outcome = match result {
-            revm::FrameResult::Call(_) | revm::FrameResult::EOFCreate(_) => unreachable!(),
-            revm::FrameResult::Create(create) => create,
-        };
+            let mut result = match first_frame_or_result {
+                revm::FrameOrResult::Frame(first_frame) => evm.run_the_loop(first_frame)?,
+                revm::FrameOrResult::Result(result) => result,
+            };
+
+            evm.handler.execution().last_frame_return(&mut evm.context, &mut result)?;
+
+            let outcome = match result {
+                revm::FrameResult::Call(_) | revm::FrameResult::EOFCreate(_) => unreachable!(),
+                revm::FrameResult::Create(create) => create,
+            };
+
+            evm.context.evm.inner.journaled_state.depth -= 1;
 
-        evm.context.evm.inner.journaled_state.depth -= 1;
+            Ok(outcome)
+        })
+    }
+
+    fn console_log<DB: DatabaseExt>(
+        &mut self,
+        ccx: &mut CheatsCtxt<DB>,
+        message: String,
+    ) -> Result<(), EVMError<DB::Error>> {
+        self.with_evm(ccx, |evm| {
+            let log =
+                Log { address: CHEATCODE_ADDRESS, data: (&Console::log { val: message }).into() };
 
-        ecx.journaled_state = evm.context.evm.inner.journaled_state;
-        ecx.env = evm.context.evm.inner.env;
-        ecx.l1_block_info = evm.context.evm.inner.l1_block_info;
-        ecx.error = evm.context.evm.inner.error;
+            evm.context.external.log(&mut evm.context.evm, &log);
```

Comment:
this can just be self.get_inspector no?

Reply:
we need to construct `revm::Evm` here to obtain `EvmContext` for `Inspector::log` call

Reply:
or you mean using `self.get_inspector` instead of `evm.context.external` in closure?

Reply:
yeah just get_inspector instead of constructing the whole evm if we're not gonna use it

Reply:
we need evm for `&mut evm.context.evm` here

`CheatsCtxt` contain only reference to `InnerEvmContext`, so we can't call inspector hooks with it

Reply:
right